File: cpp_generated/plugin.test.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/eliza/packages/project-tee-starter/__tests__/plugin.test.h"

void Main(void)
{
    spyOn(logger, std::string("info"));
    spyOn(logger, std::string("error"));
    spyOn(logger, std::string("warn"));
    describe(std::string("TEE Starter Plugin"), [=]() mutable
    {
        it(std::string("should have the correct name"), [=]() mutable
        {
            expect(teeStarterPlugin->name)->toBe(std::string("mr-tee-starter-plugin"));
        }
        );
        it(std::string("should have the correct description"), [=]() mutable
        {
            expect(teeStarterPlugin->description)->toBe(std::string("Mr. TEE's starter plugin - using plugin-tee for attestation"));
        }
        );
        it(std::string("should have no custom actions (using plugin-tee instead)"), [=]() mutable
        {
            expect(teeStarterPlugin->actions)->toEqual(array<any>());
        }
        );
        it(std::string("should have no custom providers"), [=]() mutable
        {
            expect(teeStarterPlugin->providers)->toEqual(array<any>());
        }
        );
        it(std::string("should have no custom evaluators"), [=]() mutable
        {
            expect(teeStarterPlugin->evaluators)->toBeUndefined();
        }
        );
        it(std::string("should have no custom services"), [=]() mutable
        {
            expect(teeStarterPlugin->services)->toEqual(array<any>());
        }
        );
    }
    );
}

MAIN

File: classified/packages/plugin-experience/__tests__/plugin.test.ts
import { describe, it, expect } from 'vitest';
import { experiencePlugin } from '../src/index';

describe('Experience Plugin Tests', () => {
  describe('Plugin Structure Tests', () => {
    it('should have correct plugin metadata', () => {
      expect(experiencePlugin.name).toBe('experience');
      expect(experiencePlugin.description).toBe(
        'Self-learning experience system that records experiences and learns from agent interactions'
      );
    });

    it('should export required components', () => {
      expect(experiencePlugin.services).toBeDefined();
      expect(experiencePlugin.services?.length).toBeGreaterThan(0);
      expect(experiencePlugin.providers).toBeDefined();
      expect(experiencePlugin.evaluators).toBeDefined();
    });
  });
});
